### PR TITLE
Refactor the Google Map block to make it easier

### LINKED ISSUE #495: ISBAT have a simplified Google Map experience if I don't specify an API key
We could provide a subset of the experience if a person doesn't want to register for Google Maps (iframe the map directly).

### PR DESCRIPTION
**Pull request by jonathanbardo**
Resolve #495

### CODE REVIEW

**Comment by richtabor** [comment]
New user flow, when simply adding an address:

ScreenFlow

**Comment by richtabor** [comment]
New user flow, when adding an advanced map (using a Google Maps API key): 

ScreenFlow

File: src/blocks/map/components/inspector.js [hunk 1]
```diff
@@ -197,7 +197,10 @@ class Inspector extends Component {
 							/>
 						</PanelBody>
 					}
-					<PanelBody title={ __( 'Google Maps API Key' ) } initialOpen={ false }>
+					<PanelBody
+						title={ __( 'Google Maps API Key' ) }
+						initialOpen={ address ? false : true }
```
Comment (jonathanbardo) [inline src/blocks/map/components/inspector.js]: @richtabor I think we can simplify here `initialOpen={ !address }`

File: src/blocks/map/components/controls.js [hunk 2]
```diff
@@ -26,8 +26,8 @@ function Controls( { attributes, setAttributes, apiKey } ) {
 
 	const toolbarControls = [
 		{
-			icon: 'edit',
-			title: __( 'Edit' ),
+			icon: icons.mapEdit,
```
Comment (jonathanbardo) [inline src/blocks/map/components/controls.js]: @richtabor Nice!

File: src/blocks/map/components/inspector.js (hunk 1 above)
Comment (richtabor) [reply src/blocks/map/components/inspector.js]: Good catch 👌

**Pull request merged**